cluster/v1/types.go

```diff
@@ -170,3 +170,44 @@ type ManagedClusterList struct {
 	// Items is a list of managed cluster.
 	Items []ManagedCluster `json:"items"`
 }
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ManagedClusterSet is a set of managed clusters.
+type ManagedClusterSet struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec defines the attributes of the desired managed clusters
+	Spec ManagedClusterSetSpec `json:"spec"`
+}
+
+// ManagedClusterSetSpec describes the attributes of the desired managed clusters
+type ManagedClusterSetSpec struct {
+	// Clusters represents a set of managed clusters in the cluster set
+	Clusters ClusterSelector `json:"clusterSelector,omitempty"`
+}
+
+// ClusterSelector represents a selector to select a set of managed clusters
+type ClusterSelector struct {
+	// ClusterNames represents a list of cluster name
+	ClusterNames []string `json:"clusterNames,omitempty"`
+
+	// LabelSelector represents a label selector to select cluster by label
```

Comment:
+optional

I think we need to doc what is meaning when labelSelector is empty

Reply:
Comment added.